layout/layout.cpp

```diff
@@ -99,10 +99,19 @@ int to_px(std::string_view property, int const font_size) {
         return res;
     }
 
-    if (property.ends_with("em")) {
+    auto const parsed_length = std::distance(property.data(), parse_result.ptr);
+    auto const suffix = property.substr(parsed_length);
```

Comment:
I think they call this `unit` in the CSS spec.

Reply:
Good catch! Will update the name to match the spec. Also need to add a workaround for gcc 10 not supporting from_chars with floats. :(

Reply:
Right, renamed it and added a very not spec-compliant version of from_chars for gcc 10. Can you please have a look at the last commit again?

Reply:
Also added a few tests to make sure we run a few more of the failure paths in the layout system.

Reply:
Looks good!